managed/services/backup/backup_service.go

```diff
@@ -183,7 +183,18 @@ func (s *Service) PerformBackup(ctx context.Context, params PerformBackupParams)
 		err = status.Errorf(codes.Unknown, "Unknown service: %s", svc.ServiceType)
 	}
 	if err != nil {
-		return "", err
+		var target *agents.UnsupportedAgentError
+		if errors.As(err, &target) {
+			_, dbErr := models.UpdateArtifact(s.db.Querier, artifact.ID, models.UpdateArtifactParams{
+				Status: models.BackupStatusPointer(models.ErrorBackupStatus),
+			})
+
+			if dbErr != nil {
+				s.l.WithError(err).Error("update backup artifact status")
```

Comment:
```suggestion
				s.l.WithError(err).Error("failed to update backup artifact status")
```
Otherwise it sounds like incomplete sentence